Annotate network process service worker related IPC messages

youennf opened the pull request:
#### 8521d6ff0c662b6633db27a9df949ee91c3f3616
<pre>
Annotate network process service worker related IPC messages
<a href="https://bugs.webkit.org/show_bug.cgi?id=293621">https://bugs.webkit.org/show_bug.cgi?id=293621</a>
<a href="https://rdar.apple.com/problem/152095728">rdar://problem/152095728</a>

Reviewed by Per Arne Vollan.

We update UnifiedWebPreferences.yaml to allow annotating service worker IPC messages received by networking process.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.messages.in:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.messages.in:

Canonical link: <a href="https://commits.webkit.org/295740@main">https://commits.webkit.org/295740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f27f5341c9a51dbe057422dc888dddcb721d12ba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105894 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25645 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16038 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111091 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56492 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26325 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34148 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80435 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108900 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20751 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95563 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60753 "Passed tests") | 
| | [💥 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20376 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13661 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55929 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/98535 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90135 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13697 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113941 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/104513 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33034 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24365 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89514 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33398 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91792 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89186 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22757 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34056 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11866 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28570 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32959 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38370 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/128825 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32705 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35155 "Found 1 new JSC stress test failure: wasm.yaml/wasm/gc/ref-i31-eq.js.wasm-collect-continuously (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36054 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34303 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
```diff
@@ -225,6 +225,9 @@ static inline bool isNat64IPAddress(const rtc::IPAddress& ip)
     {
         auto hostAddress = computeHostAddress(address);
+        WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+        RELEASE_LOG_ERROR(WebRTC, "youni NetworkRTCUDPSocketCocoaConnections '%s:%d'", hostAddress.c_str(), (int)address.port());
```
Should we avoid logging the full host address here?

pvollan commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
```diff
@@ -286,6 +289,8 @@ static inline bool isNat64IPAddress(const rtc::IPAddress& ip)
 
 void NetworkRTCUDPSocketCocoaConnections::setListeningPort(int port)
 {
+    RELEASE_LOG_ERROR(WebRTC, "youni NetworkRTCUDPSocketCocoaConnections::setListeningPort '%d'", port);
```
Is this logging primarily meant for debugging purposes?

youennf replied:
This was an error, will remove

pvollan commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
```diff
@@ -358,6 +363,10 @@ static inline void processUDPData(RetainPtr<nw_connection_t>&& nwConnection, Ref
 std::pair<RetainPtr<nw_connection_t>, Ref<NetworkRTCUDPSocketCocoaConnections::ConnectionStateTracker>> NetworkRTCUDPSocketCocoaConnections::createNWConnection(const rtc::SocketAddress& remoteAddress)
 {
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    RELEASE_LOG_ERROR(WebRTC, "youni NetworkRTCUDPSocketCocoaConnections::createNWConnection '%s:%d'", remoteAddress.ipaddr().ToString().c_str(), (int)remoteAddress.port());
```
Ditto.

youennf replied:
This was an error, will remove

pvollan reviewed (approved):
R=me. It seems the logging added in this patch is for debugging purposes and can be removed before landing.

youennf committed:
Annotate network process service worker related IPC messages
https://bugs.webkit.org/show_bug.cgi?id=293621
rdar://problem/152095728

Reviewed by Per Arne Vollan.

We update UnifiedWebPreferences.yaml to allow annotating service worker IPC messages received by networking process.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.messages.in:
* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.messages.in:

Canonical link: https://commits.webkit.org/295740@main
